### PR TITLE
Fix bug with false positives in unused variable rule and other problems when using language services

### LINKED ISSUE #613: no-unused-variable false positive when referenced as a type (rather than a value)
code samples below, reported in #191 

from @SenHeng:

```ts
import Foo = this.that.Foo;

module some.module.blah {
 export  class bar {
    private foo: Foo;
  }
}
```

`Foo` is a false positive

---

from @nikklassen:

```ts
import DateTimeOpts = Intl.DateTimeFormatOptions;

interface MyDateTimeOpts extends DateTimeOpts {
    timezoneOffset: number;
}

let opts: MyDateTimeOpts;
console.log(opts.timezoneOffset - 1);
```

`DateTimeOpts` is a false positive

### PR DESCRIPTION
**Pull request by JKillian**
See discussion of https://github.com/Microsoft/TypeScript/issues/4560
Fixes https://github.com/palantir/tslint/issues/570
Fixes #613

### CODE REVIEW

File: src/language/languageServiceHost.ts [hunk 1]
```diff
@@ -30,8 +30,7 @@ module Lint {
     }
 
     export function createLanguageService(fileName: string, source: string) {
-        const documentRegistry = ts.createDocumentRegistry();
```
Comment (JKillian) [inline src/language/languageServiceHost.ts]: Doing this ourselves was unnecessary as it is done internally automatically by the language service

File: src/language/languageServiceHost.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ module Lint {
             getCurrentDirectory: () => "",
             getDefaultLibFileName: () => "lib.d.ts",
             getScriptFileNames: () => [fileName],
-            getScriptSnapshot: () => ts.ScriptSnapshot.fromString(source),
+            getScriptSnapshot: name => ts.ScriptSnapshot.fromString(name === fileName ? source : ""),
```
Comment (adidahiya) [inline src/language/languageServiceHost.ts]: please wrap the function args list in parens

File: test/files/rules/nounusedvariable-falsepositives.test.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const fs = require("fs");
```
Comment (adidahiya) [inline test/files/rules/nounusedvariable-falsepositives.test.ts]: can you investigate whether this change fixes #613 and #651 as well? if so, please add those tests to this file

**Comment by JKillian** [comment]
[Made suggested changes]

File: test/files/rules/nounusedvariable-falsepositives.test.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+// case 3
+import Bar = whatever.that.Foo;
+
+module some.module.blah {
+	export class bar {
+		private bar: Bar;
+		constructor() {
+			console.log(this.bar);
+		}
+	}
```
Comment (adidahiya) [inline test/files/rules/nounusedvariable-falsepositives.test.ts]: don't mix tabs and spaces in this file; just use spaces

Comment (JKillian) [inline test/files/rules/nounusedvariable-falsepositives.test.ts]: Fixed

**Comment by adidahiya** [comment]
looks good, just waiting for CI

**Pull request merged**